### PR TITLE
feat(feed): Feed syntax enhancement

### PR DESCRIPTION
**Pull request by dr1rrb**
## Feature
Enhance feed syntax

## What is the new behavior?
* Add State.Value
* Add State.Empty
* Remove need of [BindableReactive] for VM that are not using the deprecated `IInput` pattern
* Use rolsyn generators to better support long file path

### CODE REVIEW

**Review by francoistanguay — APPROVED** [review]

File: src/Uno.Extensions.Reactive.Generator/FeedsGenerator.cs [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
+	public void Initialize(GeneratorInitializationContext context) { }
 
-	public void Execute(Uno.SourceGeneration.GeneratorExecutionContext context)
+	public void Execute(GeneratorExecutionContext context)
 	{
+#if DEBUG
+		var process = Process.GetCurrentProcess().ProcessName;
+		if (process.IndexOf("VBCSCompiler", StringComparison.OrdinalIgnoreCase) is not -1
+			|| process.IndexOf("csc", StringComparison.OrdinalIgnoreCase) is not -1)
+		{
```
Comment (carldebilly) [inline src/Uno.Extensions.Reactive.Generator/FeedsGenerator.cs:24]:
```suggestion
		if (process.Contains("VBCSCompiler", StringComparison.OrdinalIgnoreCase)
			|| process.Contains("csc", StringComparison.OrdinalIgnoreCase))
		{
```

**Review by carldebilly — APPROVED** [review]

File: src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelGenerator.cs [hunk 2]
```diff
@@ -19,10 +20,7 @@ public BindableViewModelGenerator(BindableGenerationContext ctx)
 
 	private bool IsSupported(INamedTypeSymbol type)
 		=> _ctx.IsGenerationEnabled(type)
-			?? type
-				.Constructors
-				.SelectMany(ctor => ctor.Parameters)
-				.Any(parameter => _ctx.IsInputOrCommand(parameter.Type));
+			?? type.Name.EndsWith("ViewModel", StringComparison.Ordinal) && type.IsPartial();
```
Comment (Xiaoy312) [inline src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelGenerator.cs:23]: maybe it is best to offer an [GeneratedBindableVM(true/false)] attribute to customize this?

File: src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelGenerator.cs [hunk 3]
```diff
@@ -68,76 +66,86 @@ private string Generate(INamedTypeSymbol vm)
...
-
-		return code;
+		var bindableVmCode = $@"
+			public partial class Bindable{vm.Name} : {NS.Bindings}.BindableViewModelBase
+			{{
+				{inputsErrors.Align(4)}
+				{inputs.Select(input => input.GetBackingField()).Align(4)}
+				{mappedMembers.Select(member => member.GetBackingField()).Align(4)}
+
+				{vm
```
Comment (Xiaoy312) [inline src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelGenerator.cs:77]: could maybe extract this inline block into a variable or method, to help reduce the nesting?

File: src/Uno.Extensions.Reactive.Generator/Utils/RoslynExtensions.cs [hunk 4]
```diff
@@ -109,4 +112,28 @@ public static string ToCSharpCodeString(this Accessibility accessibility)
 
 		throw new ArgumentOutOfRangeException($"{accessibility} is not supported.");
 	}
+
+	public static bool IsPartial(this INamedTypeSymbol type)
+	{
+		var syntaxRefs = type.DeclaringSyntaxReferences;
+		var isPartial = syntaxRefs.Length switch
+		{
+			0 => true,
```
Comment (Xiaoy312) [inline src/Uno.Extensions.Reactive.Generator/Utils/RoslynExtensions.cs:121]: what is the reasoning behind the 0 case?

File: src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelGenerator.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelGenerator.cs:23]: It's the case just above. The "end by ViewModel" is the general case, but you do have an attribute to customize top match your needs.

File: src/Uno.Extensions.Reactive.Generator/Utils/RoslynExtensions.cs (hunk 4 above)
Comment (dr1rrb) [reply src/Uno.Extensions.Reactive.Generator/Utils/RoslynExtensions.cs:121]: Depending of when your generator runs, you can have some code that is not compiling and roslyn was able to determine that it should be a class type, but was not able yet where the class was defined. It should not occur in our case, but it's always better handle all cases!

File: src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelGenerator.cs (hunk 3 above)
Comment (dr1rrb) [reply src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelGenerator.cs:77]: Well is not necessarily the most annoying part here ... I tried to do like in a T4 file where you generate the code inline, higher nesting, but you can almost read the generated code just by reading the file. I would say that there is no real good pattern and it's more a personal choice in that case.

**Pull request merged**